python/taichi/lang/impl.py

```diff
@@ -138,6 +139,26 @@ def chain_compare(comparators, ops):
     return ret
 
 
+@taichi_scope
+def func_call(func, *args, **kwargs):
+    _taichi_skip_traceback = 1
```

Comment:
Is the purpose of this function to check if `func` is decorated by `@ti.func`? Is so , can we add a special mark to `decorated` returned from `@ti.func`, e.g. `_is_taichi_func`, and check that special mark here? This avoids all these special handlings. As an example, `ti.data_oriented` is doing something like this: https://github.com/taichi-dev/taichi/blob/6dc41cee29599004bc0a74223b740fba096779eb/python/taichi/lang/kernel.py#L604

Reply:
Good idea, let's add `_func_impl` like `_kernel_impl` iapr.